core/pkg/ingress/controller/launch.go

```diff
@@ -234,6 +244,13 @@ func createApiserverClient(apiserverHost string, kubeConfig string) (*client.Cli
 	if err != nil {
 		return nil, err
 	}
+
+	// test connection
+	_, err = client.ServerVersion()
```

Comment:
Why are you doing this?
If is not possible to reach the api server [this error](https://github.com/kubernetes/ingress/blob/master/core/pkg/ingress/controller/launch.go#L108) is returned.
In any other case the client will retry the connection.